modules/jfx.incubator.richtext/src/main/java/jfx/incubator/scene/control/richtext/model/BasicTextModel.java

```diff
@@ -0,0 +1,293 @@
+         *
+         * @param index the paragraph index
+         * @param offset the insertion offset within the paragraph
+         * @param text the text to insert
+         * @param attrs the style attributes
+         * @return the number of characters inserted
+         */
+        public int insertTextSegment(int index, int offset, String text, StyleAttributeMap attrs);
+
+        /**
+         * Inserts a line break.
+         *
+         * @param index the model index
+         * @param offset the text offset
+         */
+        public void insertLineBreak(int index, int offset);
+
+        /**
+         * This method gets called only if the model is editable.
+         * The caller guarantees that {@code start} precedes {@code end}.
+         *
+         * @param start the start of the region to be removed
+         * @param end the end of the region to be removed, expected to be greater than the start position
+         */
+        public void removeRange(TextPos start, TextPos end);
+
+        /**
+         * Determines whether this content supports modification by the user.
+         * @return true if editable
+         */
+        public boolean isUserEditable();
+    }
+
+    private final Content content;
+
+    /**
+     * Constructs an empty model with the specified {@code Content}.
+     * @param c the content to use
+     */
+    public BasicTextModel(Content c) {
+        this.content = c;
+        registerDataFormatHandler(PlainTextFormatHandler.INSTANCE, true, true, 0);
+    }
+
+    /**
+     * Constructs an empty model with the in-memory {@code Content}.
+     */
+    public BasicTextModel() {
+        this(new InMemoryContent());
+    }
+
+    /**
+     * Inserts text at the specified position.
+     * This is a convenience shortcut for {@link #replace(StyleResolver, TextPos, TextPos, String, boolean)}.
+     *
+     * @param p the insertion position
+     * @param text the text to insert
+     * @throws NullPointerException if the model is {@code null}
+     * @throws UnsupportedOperationException if the model is not {@link StyledTextModel#isWritable() writable}
+     */
+    public void insertText(TextPos p, String text) {
+        replace(null, p, p, text, false);
+    }
+
+    @Override
+    public int size() {
+        return content.size();
+    }
+
+    @Override
+    public String getPlainText(int index) {
+        return content.getText(index);
+    }
+
+    @Override
+    public RichParagraph getParagraph(int index) {
+        String text = getPlainText(index);
+        return RichParagraph.builder().
+            addSegment(text).
+            build();
+    }
+
+    /**
+     * Determines whether the model is user-editable.
+     * <p>
+     * This method calls {@link BasicTextModel.Content#isUserEditable()}.
+     *
+     * @return true if the model is user-editable
+     */
+    @Override
+    public final boolean isWritable() {
+        return content.isUserEditable();
```

Comment:
Just curious why it is editable in some places but writable in other? Again, description says user-editable name says writable.

Reply:
changed to writable.